experimental/packages/opentelemetry-sdk-metrics-base/src/MeterProvider.ts

```diff
@@ -38,8 +38,8 @@ export class MeterProvider {
   private _metricReaders: MetricReader[] = [];
   private _metricExporters: MetricExporter[] = [];
 
-  constructor(options: MeterProviderOptions) {
-    this._sharedState = new MeterProviderSharedState(options.resource ?? Resource.empty());
+  constructor(options?: MeterProviderOptions) {
+    this._sharedState = new MeterProviderSharedState(options?.resource ?? Resource.empty());
   }
 
   getMeter(name: string, version = '', options: metrics.MeterOptions = {}): metrics.Meter {
```

Comment:
If the schema URL is part of the identity, shouldn't it be a top-level property and not buried in options?

Reply:
Actually, there is only one requirement for meter creation:

> Implementations MUST NOT require users to repeatedly obtain a Meter again with the same name+version+schema_url to pick up configuration changes. This can be achieved either by allowing to work with an outdated configuration or by ensuring that new configuration applies also to previously returned Meters.
> https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#get-a-meter

It also suggests that:

> It is unspecified whether or under which conditions the same or different Meter instances are returned from this function.

So, as long as we propagate the configurations to all meters created without another `getMeter` call, it should be spec-compliant.

As we are using the `MeterProviderSharedState` pattern to share the meter provider configurations, I'd believe we've already met the bar that the created meters can automatically get configuration updates without the need to call `getMeter` again. So it seems that we actually don't have to return an identical meter for each `getMeter` call.

Reply:
Another problem is that the spec is said instrument names are unique in a meter:

> Meter implementations MUST return an error when multiple Instruments are registered under the same Meter instance using the same name.
> https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument

So, if we are going to return a new meter instance for every `getMeter`, how do we define "same Meter instance" in the above context?

Reply:
Submitted a spec issue for this: https://github.com/open-telemetry/opentelemetry-specification/issues/2226